clients/src/main/java/org/apache/kafka/clients/consumer/NoOffsetForPartitionException.java

```diff
@@ -29,19 +31,24 @@
 
     private static final long serialVersionUID = 1L;
 
-    private final TopicPartition partition;
+    private final Set<TopicPartition> partitions;
 
     public NoOffsetForPartitionException(TopicPartition partition) {
         super("Undefined offset with no reset policy for partition: " + partition);
-        this.partition = partition;
+        this.partitions = Collections.singleton(partition);
+    }
+
+    public NoOffsetForPartitionException(Collection<TopicPartition> partitions) {
+        super("Undefined offset with no reset policy for partitions: " + partitions);
+        this.partitions = Collections.unmodifiableSet(new HashSet<>(partitions));
     }
 
     public TopicPartition partition() {
```

Comment:
Maybe we should deprecate this method in favour of `partitions` with an explanation.